ipam/allocator_test.go

```diff
@@ -987,6 +976,122 @@ func TestRequest(t *testing.T) {
 	}
 }
 
+// TestOverlappingRequests tests that overlapping subnets cannot be allocated.
+// Requests for subnets which are supersets or subsets of existing allocations,
+// or which overlap at the beginning or end, should not be permitted.
+func TestOverlappingRequests(t *testing.T) {
+	input := []struct {
+		environment []string
+		subnet      string
+		ok          bool
+	}{
+		// Ok: Non-overlapping left and right
+		{[]string{"10.0.0.0/8"}, "11.0.0.0/8", true},
```

Comment:
I think some of these lists can be compressed because they test the same aspect of it.

Reply:
for eg ```{[]string{"10.0.0.0/8"}, "11.0.0.0/8", true},``` and ```{[]string{"10.0.0.0/8"}, "9.0.0.0/8", true},``` are pretty much same and so are the rest of rows in "ok"  .